### ISSUE TITLE
Bad USING with locals?

### ISSUE THREAD

**Issue by mrjbq7** [slavapestov/factor#1378]
This definition:

```factor
:: foo ( seq -- seq' ) seq [| x | x 1 + ] map ;
```

Seems to ``see`` with a USING of ``locals.prettyprint`` which isn't actually needed?

```factor
USING: locals locals.prettyprint math sequences ;
IN: scratchpad
:: foo ( seq -- seq' ) seq [| x | x 1 + ] map ;
```

**Comment by mrjbq7** [comment]
Oh, ``|`` is in ``locals.prettyprint``.

**Comment by mrjbq7** [comment]
So, it's not needed when parsing, but seems to be needed when printing.

**Comment by erg** [comment]
What's the status of this one?

**Comment by mrjbq7** [comment]
It's a problem when prettyprinting ``|``, it needs it to be defined as a ``SYMBOL:``, so it makes the ``USING:`` require ``locals.prettyprint``.

One option to fix this would be to move the definition of ``|`` into ``locals``...?

**Comment by erg** [comment]
We don't want to have to include the prettyprinter whenever we use locals.
Is there a way not to have | as a symbol maybe?

**Comment by mrjbq7** [comment]
This fixes it, the ``|`` is not clickable:

```diff
diff --git a/basis/locals/prettyprint/prettyprint.factor b/basis/locals/prettypr
int/prettyprint.factor
index b0fbebb..7c3781e 100644
--- a/basis/locals/prettyprint/prettyprint.factor
+++ b/basis/locals/prettyprint/prettyprint.factor
@@ -20,7 +20,7 @@ M: lambda pprint*
     <flow
     \ [| pprint-word
     dup vars>> pprint-vars
-    \ | pprint-word
+    "|" text
     f <inset body>> pprint-elements block>
     \ ] pprint-word
     block> ;
```

Want me to commit?

**Comment by erg** [comment]
Sure! ``|`` is a dumb word anyway. Can we entirely remove it?

**Comment by mrjbq7** [comment]
Okay, fixed in ec7afb3ea65c51d82ccf65bce6d42a5aa31b7662!
